containers/src/Kokkos_DynRankView.hpp

```diff
@@ -389,25 +408,57 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
   template <class, class...>
   friend class Kokkos::Impl::ViewMapping;
 
+  size_t m_rank{};
+
+  using drdtraits = Impl::DynRankDimTraits<void>;
+
  public:
   using drvtraits = ViewTraits<DataType, Properties...>;
 
   using view_type = View<DataType*******, Properties...>;
 
-  using traits = ViewTraits<DataType*******, Properties...>;
-
- private:
-  using map_type =
-      Kokkos::Impl::ViewMapping<traits, typename traits::specialize>;
-  using track_type = Kokkos::Impl::SharedAllocationTracker;
-
-  track_type m_track;
-  map_type m_map;
-  unsigned m_rank;
+  // typedefs from ViewTraits, overriden
+  using data_type           = typename drvtraits::data_type;
+  using const_data_type     = typename drvtraits::const_data_type;
+  using non_const_data_type = typename drvtraits::non_const_data_type;
+
+  // typedefs from ViewTraits not overriden
+  using value_type           = typename view_type::value_type;
+  using const_value_type     = typename view_type::const_value_type;
+  using non_const_value_type = typename view_type::non_const_value_type;
+  using traits               = typename view_type::traits;
+  using array_layout         = typename view_type::array_layout;
+
+  using execution_space = typename view_type::execution_space;
+  using memory_space    = typename view_type::memory_space;
+  using device_type     = typename view_type::device_type;
+
+  using memory_traits     = typename view_type::memory_traits;
+  using host_mirror_space = typename view_type::host_mirror_space;
+  using size_type         = typename view_type::size_type;
+
+  using reference_type = typename view_type::reference_type;
+  using pointer_type   = typename view_type::pointer_type;
+
+  using scalar_array_type           = value_type;
+  using const_scalar_array_type     = const_value_type;
+  using non_const_scalar_array_type = non_const_value_type;
+  using specialize                  = typename view_type::specialize;
+
+  // typedefs in View for mdspan compatibility
+  // cause issues with MSVC+CUDA
+  // using layout_type  = typename view_type::layout_type;
+  using index_type       = typename view_type::index_type;
+  using element_type     = typename view_type::element_type;
+  using rank_type        = typename view_type::rank_type;
+  using reference        = reference_type;
+  using data_handle_type = pointer_type;
 
- public:
   KOKKOS_INLINE_FUNCTION
   view_type& DownCast() const { return (view_type&)(*this); }
+  KOKKOS_FUNCTION
+  view_type to_view() const { return *this; }
```

Comment:
Is this a new function? Why do we need/want it?

Reply:
this should replace `DownCast`, but I gonna comment it out for now with a FIXME message ok? `DownCast` doesn't follow our naming convention, and `to_view` corresponds to the `to_mdspan` we introduced.

Reply:
commented the function out and added FIXME comments.